haystack/document_stores/faiss.py

```diff
@@ -133,6 +149,31 @@ def __init__(
             index=index
         )
 
+        self._validate_index_sync()
+
+    def _validate_params_load_from_disk(self, sig: Signature, locals: dict, kwargs: dict):
+        allowed_params = ["faiss_index_path", "faiss_config_path", "self", "kwargs"]
+        invalid_param_set = False
+
+        for param in sig.parameters.values():
+            if param.name not in allowed_params:
+                if param.default != locals[param.name]:
+                    invalid_param_set = True
+                    break
+        
```

Comment:
Another style detail: the loop can be summarized in a condition with `any`. It's long though, so I'll leave up to you wnat's more readable :sweat_smile: 
```python
params_are_invalid = any(param.name not in allowed_params and param.default != locals[param.name] for param in sig.parameters.values())
```

Reply:
Thanks for spotting this, I also thought about the any clause because I really like comprehensions. But I opted for the loop for line length and understandability.